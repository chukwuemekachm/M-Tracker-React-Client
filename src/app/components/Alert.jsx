import React from 'react';
import PropTypes from 'prop-types';

const Alert = ({ message, backgroundColor }) => (
  <div className="ch-alert" style={{ backgroundColor }}>
    <p>
      {message}
    </p>
  </div>
);

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
};

export default Alert;
