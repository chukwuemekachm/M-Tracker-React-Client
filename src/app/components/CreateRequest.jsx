import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../assets/css/createRequestForm.css';

/**
 * @description Create request form modal component
 *
 * @class
 *
 * @extends React.Component
 */
class CreateRequestForm extends Component {
  /**
   * @constructor
   *
   * @param {object} props The injected props
   */
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      type: '',
      description: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  /**
   * @description Handles the input change events
   * Updates the state value of the input field
   *
   * @function
   *
   * @param {object} event The event object
   */
  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  /**
   * @description Handles the form submit event
   * Executes the create method to submit the form
   * Handle the result of the response
   *
   * @function
   *
   * @param {object} event The event object
   */
  handleSubmit(event) {
    event.preventDefault();
    const { create, history } = this.props;
    create(this.state).then((response) => {
      if (response) {
        this.setState({
          title: '',
          type: '',
          description: '',
        });
        switch (response.code) {
          case 201:
            document.getElementById('closeCreateModal').click();
            history.push('/dashboard');
            break;
          case 401:
            history.push('/login');
            break;
          default:
            break;
        }
      }
    });
  }

  /**
   * @description Renders the component on a dom node
   *
   * @function
   *
   * @returns {object} The components to render
   */
  render() {
    const { title, type, description } = this.state;
    return (
      <div
        className="modal fade"
        id="createRequest"
        tabIndex="-1"
        role="dialog"
        aria-labelledby="createRequest"
        aria-hidden="true"
      >
        <div className="modal-dialog modal-dialog-centered" role="document">
          <div className="modal-content">
            <div className="modal-header p-2 w3-flat-midnight-blue">
              <h5 className="modal-title" id="exampleModalCenterTitle">
                Create a new request
              </h5>
              <button
                type="button"
                id="closeCreateModal"
                className="close p-4"
                data-dismiss="modal"
                aria-label="Close"
              >
                <span aria-hidden="true">
                  &times;
                </span>
              </button>
            </div>
            <div className="modal-body">
              <form onSubmit={this.handleSubmit}>
                <div className="input-group mb-4">
                  <input
                    type="text"
                    className="form-control text"
                    placeholder="Title"
                    name="title"
                    onChange={this.handleChange}
                    value={title}
                    required
                  />
                </div>
                <br />
                <div className="input-group mb-4">
                  <select
                    name="type"
                    className="form-control text"
                    onChange={this.handleChange}
                    required
                    value={type}
                  >
                    <option value="">
                      Type
                    </option>
                    <option value="repair">
                      Repair
                    </option>
                    <option value="maintenance">
                      Maintenance
                    </option>
                  </select>
                </div>
                <br />
                <div className="input-group mb-4">
                  <textarea
                    name="description"
                    className="form-control text"
                    placeholder="Description"
                    onChange={this.handleChange}
                    value={description}
                    required
                  />
                </div>
                <div className="p-4">
                  <button type="submit" className="btn btn-success btn-md btn-block">
                    Create request
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

CreateRequestForm.propTypes = {
  create: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
};


export default CreateRequestForm;
