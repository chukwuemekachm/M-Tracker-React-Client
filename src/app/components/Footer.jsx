import React from 'react';
import '../assets/css/footer.css';

/**
 * @description The Footer component
 *
 * @returns {object}
 */
const Footer = () => (
  <div className="w3-flat-midnight-blue footer">
    Copyright of &copy;2018 Maintenance Tracker by Chima Chukwuemeka
  </div>
);

export default Footer;
