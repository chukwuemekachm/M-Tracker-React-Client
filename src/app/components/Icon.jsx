import React from 'react';
import PropTypes from 'prop-types';
import '../assets/css/icon.css';

const Icon = ({ title, icon }) => (
  <div className="col-md-3 text-center">
    <div className="icon-container">
      <img src={icon} className="about-img" alt="authenticate" />
    </div>
    <h4>
      {title}
    </h4>
  </div>
);

Icon.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};

export default Icon;
