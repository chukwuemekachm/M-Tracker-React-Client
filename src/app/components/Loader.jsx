import React from 'react';
import loadingSvg from '../assets/svg/loading.svg';
import '../assets/css/loader.css';

const Loader = () => (
  <div className="loader-container">
    <div className="loader">
      <img src={loadingSvg} alt="loader" />
    </div>
  </div>
);

export default Loader;
