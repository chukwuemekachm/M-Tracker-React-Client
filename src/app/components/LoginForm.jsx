import React, { Component } from 'react';
import PropTypes from 'prop-types';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  onSubmit(event) {
    event.preventDefault();

    const { userLoginRequest, history } = this.props;
    userLoginRequest(this.state, history, 'login');
  }

  render() {
    const {
      email, password,
    } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <label htmlFor="email" className="ch-label">
          Email Address
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="email"
            name="email"
            className="ch-input"
            required
            placeholder="Email Address"
            value={email}
            onChange={this.onChange}
          />
        </label>
        <label htmlFor="password" className="ch-label">
          Password
          {' '}
          <span className="ch-require">
            *
          </span>
          <input
            type="password"
            name="password"
            className="ch-input"
            required
            placeholder="Password"
            value={password}
            onChange={this.onChange}
          />
        </label>
        <button type="submit" className="ch-btn-flat">
          Login
        </button>
      </form>
    );
  }
}
LoginForm.propTypes = {
  userLoginRequest: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};

export default LoginForm;
