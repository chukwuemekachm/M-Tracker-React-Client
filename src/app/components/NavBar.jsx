import React from 'react';

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      size: '',
    };
    this.toggleMenu = this.toggleMenu.bind(this);
  }

  toggleMenu() {
    let { size } = this.state;
    size = size === 'none' ? 'block' : 'none';
    this.setState({ size });
  }

  render() {
    const { size } = this.state;
    return (
      <nav>
        <ul className="ch-nav ch-card">
          <li>
            <h2>
              <a href="/">
                <i className="icon ion-md-construct" />
                {' '}
                Maintenance Tracker
              </a>
              <i className="icon ion-md-menu" id="menu" onClick={this.toggleMenu} />
            </h2>
          </li>
          <div className="ch-nav-right" id="nav" style={{ display: size }}>
            <li>
              <a href="/">
                Login
              </a>
            </li>
          </div>
        </ul>
      </nav>
    );
  }
}

export default NavBar;
