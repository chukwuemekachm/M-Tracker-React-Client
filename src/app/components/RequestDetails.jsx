import React from 'react';
import PropTypes from 'prop-types';

const RequestDetails = ({
  actions, title, type, description, status, createdat,
}) => (
  <div className="ch-row">
    <div className="ch-col-12">
      <h3 className="ch-display" dangerouslySetInnerHTML={actions} />
      <p className="ch-display">
        <label htmlFor="title" className="ch-label">
          <h4>
            Title
          </h4>
        </label>
      </p>
      <p className="ch-display">
        {title}
      </p>
      <p className="ch-display">
        <label htmlFor="type" className="ch-label">
          <h4>
            Type
          </h4>
        </label>
      </p>
      <p className="ch-display">
        {type}
      </p>
      <p className="ch-display">
        <label htmlFor="description" className="ch-label">
          <h4>
            Description
          </h4>
        </label>
      </p>
      <p className="ch-display">
        {description}
      </p>
      <p className="ch-display">
        <label htmlFor="status" className="ch-label">
          <h4>
            Status
          </h4>
        </label>
      </p>
      <p className="ch-display">
        {status}
      </p>
      <p className="ch-display">
        <label htmlFor="date" className="ch-label">
          <h4>
            Date
          </h4>
        </label>
      </p>
      <p className="ch-display">
        {new Date(createdat).toLocaleString('en-GB', { hour12: true })}
      </p>
    </div>
  </div>
);

RequestDetails.propTypes = {
  actions: PropTypes.shape({}).isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  createdat: PropTypes.string.isRequired,
};

export default RequestDetails;
