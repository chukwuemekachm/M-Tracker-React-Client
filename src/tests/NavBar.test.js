import React from 'react';
import { shallow } from 'enzyme';
import { NavBar } from '../app/components/NavBar';

describe('Navbar component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<NavBar authenticated />);
    expect(wrapper).toMatchSnapshot();
  });
});
