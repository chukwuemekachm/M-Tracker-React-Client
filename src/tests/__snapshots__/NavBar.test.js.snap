// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render without errors 1`] = `
<nav
  className="navbar navbar-expand-lg w3-flat-midnight-blue"
>
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    <i
      className="icon ion-md-construct"
    />
     
    Maintenance Tracker
  </Link>
  <button
    aria-controls="navbarHidden"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarHidden"
    data-toggle="collapse"
    type="button"
  >
    <i
      className="icon ion-md-menu w3-text-white"
    />
  </button>
  <div
    className="collapse navbar-collapse justify-content-end"
    id="navbarHidden"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <Link
          replace={false}
          to="/"
        >
          <button
            className="btn btn-success"
            type="button"
          >
            Logout
          </button>
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
